engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
     * 2.) Raw inputs and it's coercion
     * 3.) Workflow Options
     * 4.) RuntimeAttributes w.r.t. a backend
+    *
     * @return The result of the validation as instance of `ValidationResult`
     */
-  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = {
-    (for {
-      namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      validatedInputs <- Future(validateInputs(workflowInputs, namespaceWithWorkflow))
-      validatedWorkflowOptions <- Future(validateWorkflowOptions(workflowOptions, CromwellBackend.backend()))
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
```

Comment:
Yeah Jeff pretty much summed up my concerns.  If all this actor does is spin off Futures the parallelism is essentially out of control. Eventually this will hit a limit when the threads in this actor's `ExecutionContext` have been exhausted. But if you're being bottlenecked by a single validation actor wouldn't it be better to have the topmost validation actor act as a router to worker validation actors and use knowledge of Cromwell's current state to decide whether to elastically bring up more workers or simply push back on requests?

More parallelism in arbitrary parts of the application is not necessarily better.  The throughputs of the validation and execution stages will likely vary with configuration, so it would be good if the elasticities in these stages responded in reasonable ways with respect to one another rather than always just going as wide as possible.

I have broader issues with the way Cromwell currently validates and executes workflows, written up [here](#564).  While I don't currently think refactoring along these lines is strictly required for pluggable backends, it could certainly help with the issues we've already seen where Cromwell is not well behaved under heavy submission loads.